sky/packages/sky/lib/src/fn3/drag_target.dart

```diff
@@ -103,21 +101,113 @@ class DragController {
     _activeTargetWillAcceptDrop = _activeTarget != null && _activeTarget.didEnter(data);
   }
 
-  void cancel() {
-    if (_activeTarget != null)
-      _activeTarget.didLeave(data);
-    _activeTarget = null;
-    _activeTargetWillAcceptDrop = false;
+  DragTargetState _getDragTarget(List<HitTestEntry> path) {
+    // TODO(abarth): Why do we reverse the path here?
+    for (HitTestEntry entry in path.reversed) {
+      if (entry.target is RenderMetaData) {
+        RenderMetaData renderMetaData = entry.target;
+        if (renderMetaData.metaData is DragTargetState)
+          return renderMetaData.metaData;
+      }
+    }
+    return null;
   }
 
-  void drop() {
-    if (_activeTarget == null)
-      return;
-    if (_activeTargetWillAcceptDrop)
-      _activeTarget.didDrop(data);
-    else
-      _activeTarget.didLeave(data);
+  void popped([DragEndKind endKind]) {
+    if (_activeTarget != null) {
+      if (endKind == DragEndKind.dropped && _activeTargetWillAcceptDrop)
+        _activeTarget.didDrop(data);
+      else
+        _activeTarget.didLeave(data);
+    }
     _activeTarget = null;
     _activeTargetWillAcceptDrop = false;
+    if (onDragFinished != null)
+      onDragFinished();
+    super.popped(endKind);
+  }
+
+  bool get ephemeral => true;
+  bool get modal => false;
+
+  Duration get transitionDuration => const Duration();
+  bool get isOpaque => false;
+  Widget build(Key key, NavigatorState navigator) {
+    return new Positioned(
+      left: _lastOffset.dx,
+      top: _lastOffset.dy,
+      child: new IgnorePointer(
+        child: new Opacity(
+          opacity: 0.5,
```

Comment:
Show we have the opinion about the opacity or push that down into the feedback widget?